tests/unit/modules/test_hanamod.py

```diff
@@ -0,0 +1,616 @@
+
+# -*- coding: utf-8 -*-
+'''
+    :codeauthor: Xabier Arbulu Insausti <xarbulu@suse.com>
+'''
+
+# Import Python Libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+from salt import exceptions
+
+# Import Salt Testing Libs
+from tests.support.mixins import LoaderModuleMockMixin
+from tests.support.unit import TestCase, skipIf
+from tests.support import mock
+from tests.support.mock import (
+    MagicMock,
+    patch,
+    NO_MOCK,
+    NO_MOCK_REASON
+)
+
+# Import Salt Libs
+import salt.modules.hanamod as hanamod
+
+
+@skipIf(NO_MOCK, NO_MOCK_REASON)
+class HanaModuleTest(TestCase, LoaderModuleMockMixin):
+    '''
+    This class contains a set of functions that test salt.modules.hana.
+    '''
+
+    def setup_loader_modules(self):
+        return {hanamod: {}}
+
+    @patch('salt.modules.hanamod.hana.HanaInstance')
+    def test_init_return(self, mock_hana):
+        '''
+        Test _init method
+        '''
+        mock_hana_inst = MagicMock()
+        mock_hana.return_value = mock_hana_inst
+        hana_inst = hanamod._init('prd', '00', 'pass')
+        mock_hana.assert_called_once_with('prd', '00', 'pass')
+        self.assertEqual(mock_hana_inst, hana_inst)
```

Comment:
@isbm I'm changing all unittest assertion and I have noticed something that I don't like.
If I run the tests using tests/runtests.py (recommended by salt) if the assertion fails, the shown error message is really poor (only AssertionError). However, if I run the tests with py.test directly, the message is far more better.
As the official way of running the tests is the first one, this disturbs me. 
What do you think about that?

Reply:
By the way, reviewing again the salt documentation (https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) unittest assertion are always used. I don't know If the docs are not updated, or they should be re-checked...

Reply:
In fact, test_zypper.py itself uses unittest assertions... I'm a bit confused here

Reply:
@arbulu89 This is also depends on the pytest configuration. I usually blowing away whatever SaltStack configuration is :wink: because it works properly only for them (and this is sad) and using just default. Then you run your unit test as:

```
pytest --disable-warnings --tb=native --color=yes -svv /your/test_case.py <ENTER>
```

Reply:
Ah, their runtests is something I always ignore and never use. It is an old stuff that went through many political preferences overtime, is buggy and has weird configuration. There is no really official way, but "outdated way". SaltStack even doesn't tell anyone they actually running PyTest under the hood, and this is why there are plenty try/except on pytest import, which is no longer needed as of today. :wink: They supposed to annnounce that in 2017, but seems forgot or something.